.github/workflows/CI2.yml

```diff
@@ -0,0 +1,90 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron:  '0 3 * * *'
+
+jobs:
+  CI:
+    continue-on-error: ${{ matrix.kokkos_ver == 'develop' || matrix.distro == 'fedora:latest' || matrix.distro == 'fedora:intel' || matrix.distro == 'ubuntu:devel' }}
+    strategy:
+      matrix:
+        distro: ['fedora:latest', 'fedora:rawhide', 'opensuse:latest', 'ubuntu:latest', 'ubuntu:rolling'] # 'ubuntu:devel'
+        cxx: ['g++', 'clang++']
+        openmp: ['ON']
+        cmake_build_type: ['Debug', 'Release']
+        kokkos_ver: ['3.2.01', 'master', 'develop']
+        include:
+          - distro: 'fedora:intel'
+            cxx: 'icpc'
+            openmp: 'ON'
+            cmake_build_type: 'Release'
+            kokkos_ver: 'master'
+          - distro: 'fedora:intel'
+            cxx: 'icpc'
+            openmp: 'ON'
+            cmake_build_type: 'Debug'
+            kokkos_ver: 'master'
+          - distro: 'fedora:intel'
+            cxx: 'icpc'
+            cmake_build_type: 'Release'
+            openmp: 'OFF'
+            kokkos_ver: 'master'
+            cxx_flags: ' -parallel'
+          - distro: 'fedora:intel'
+            cxx: 'icpc'
+            openmp: 'OFF'
+            cmake_build_type: 'Debug'
+            kokkos_ver: 'master'
+            cxx_flags: ' -parallel'
+    runs-on: ubuntu-20.04
+    container: ghcr.io/kokkos/ci-containers/${{ matrix.distro }}
+    steps:
+      - name: Get trail license
+        if: ${{ matrix.cxx == 'icpc' }}
+        run: |
+          mkdir ~/Licenses
+          curl https://dynamicinstaller.intel.com/api/v2/license > ~/Licenses/intel.lic
+      - name: Checkout kokkos
+        uses: actions/checkout@v2.2.0
+        with:
+          repository: kokkos/kokkos
+          ref: ${{ matrix.kokkos_ver }}
+          path: kokkos
+      - name: Build kokkos
+        working-directory: kokkos
+        run: |
+          cmake -B build \
+            -DCMAKE_INSTALL_PREFIX=$HOME/kokkos \
+            -DKokkos_ENABLE_OPENMP=${{ matrix.openmp }} \
+            -DCMAKE_CXX_COMPILER=${{ matrix.cxx }} \
+            -DKokkos_ENABLE_HWLOC=ON \
+            -DCMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }}
+          cmake --build build --parallel 2
+          cmake --install build
+      - name: Checkout Cabana
+        uses: actions/checkout@v2.2.0
+      - name: Build Cabana
+        run: |
+          cmake -B build \
+            -DCMAKE_INSTALL_PREFIX=$HOME/Cabana \
+            -DMPIEXEC_MAX_NUMPROCS=2 -DMPIEXEC_PREFLAGS="--oversubscribe" \
```

Comment:
Is 2 MPI ranks our limit here? Do we just get one physical core per build? Would be nice to do 4 if possible - even if we oversubscribe in terms of MPI.

Reply:
It must be one if we couldn't run without this. Maybe change that separately?

Reply:
Yeah, I think github actions has 4 cores, but let's bump that in an extra PR.